apps/services/user-notification/src/app/modules/notifications/notificationsWorker/notificationsWorker.service.ts

```diff
@@ -320,10 +330,54 @@ export class NotificationsWorkerService implements OnApplicationBootstrap {
           message,
         }
 
-        await Promise.all([
-          this.handleDocumentNotification(handleNotificationArgs),
-          this.handleEmailNotification(handleNotificationArgs),
-        ])
+        // should always send email notification
+        const notificationPromises = [
+          void this.handleEmailNotification(handleNotificationArgs),
+        ]
+
+        // If the message is not on behalf of anyone, we look up delegations for the recipient and add messages to the queue for each delegation
+        if (!message.onBehalfOf) {
+          const shouldSendEmailToDelegations =
+            await this.featureFlagService.getValue(
+              Features.shouldSendEmailNotificationsToDelegations,
+              false,
+              { nationalId: message.recipient } as User,
+            )
+
+          if (shouldSendEmailToDelegations) {
+            // don't fail if we can't get delegations
+            try {
+              const delegations =
+                await this.delegationsApi.delegationsControllerGetDelegationRecords(
+                  {
+                    xQueryFromNationalId: message.recipient,
+                    scope: DocumentsScope.main,
+                  },
+                )
+
+              await Promise.all(
+                delegations.data.map((delegation) =>
+                  this.queue.add({
+                    ...message,
+                    recipient: delegation.toNationalId,
+                    onBehalfOf: message.recipient,
+                  }),
+                ),
+              )
+            } catch (error) {
+              this.logger.error('Error adding delegations to message queue', {
+                error,
+              })
+            }
+          }
+
+          // Only send push notifications for the main recipient
+          notificationPromises.push(
+            void this.handleDocumentNotification(handleNotificationArgs),
+          )
```

Comment:
~ I would move this to the top of the `if (!message.onBehalfOf)` statement to fire this immediately before doing the delegation lookup.